wordpress-mobile/WordPress-iOS
# Issues/4093 swap lists

This PR adds the final touches on the refreshed reader post list and cards. 
Closes #4093, #4032
For comparison the prototypes are here: https://marvelapp.com/66gi8f#7304611 

Changes include:
- The old list is deleted.
- The new post list is used everywhere the old one was previously used. 
- Headers for tags, lists and sites are no longer placeholders
- Following/Unfollowing from tag and site headers is enabled
- Tapping a tag will preview that tags posts in a new list.
- The logged out reader does not show/enable features that require a login, e.g. likes, blocking, following.

To test:
Menu:
- Test the menu when logged out and when logged in. 
- The first section is default topics and has not title. There should be a section for tags, and may or may not be a section for lists depending on if you're following any. 
- Confirm following a topic still works via the search bar.
- Confirm following a site still works via the search bar. 
- Confirm tags can still be unfollowed via the menu
- Confirm sites can still be unfollowed via the menu

Default topics (freshly pressed, blogs I follow)
- Ensure these topics do not show a header.

Tags
- Ensure tags show a tag header with the tag name and follow button. 
- Test following / unfollowing via the header. 

Lists
- Ensure lists show a list header with the name of the list. 

Sites
- Ensure sites show a header that includes the blavatar, site name, url, subscriber count and a description (if there is one). 
- Test following / unfollowing via the header.

General:
- Test pull to refresh & infinite scroll
- Tap the name of a site in a post card to preview its site. Confirm that the site name is disabled in the cards for the preview
- Tap to preview a tag.
- Confirm you can like and unlike a post
- Confirm you can view comments for a post
- Confirm you can block a site via the ellipsis menu
- Confirm you can share a post via the ellipsis menu


@koke would you be up for a review of this one? 

Needs Review: @koke
